pr_agent/tools/pr_similar_issue.py

```diff
@@ -106,8 +102,7 @@ def __init__(self, issue_url: str, args: list = None):
 
     async def run(self):
         get_logger().info('Getting issue...')
-        repo_name, original_issue_number = self.git_provider._parse_issue_url(self.issue_url.split('=')[-1])
-        issue_main = self.git_provider.repo_obj.get_issue(original_issue_number)
+        issue_main, original_issue_number = self.git_provider.get_issue(self.issue_url.split('=')[-1])
         issue_str, comments, number = self._process_issue(issue_main)
```

Comment:
**Suggestion:** Consider using a helper function to process the issue, as this operation is repeated multiple times in the code. This will enhance code readability and maintainability. [maintainability]